pkg/resourcesets/collector.go

```diff
@@ -248,157 +174,147 @@ func (h *ResourceHandler) filterByLabel(ctx context.Context, dr dynamic.Resource
 	return resourceObjectsList, nil
 }
 
+func (h *ResourceHandler) filterByKind(filter v1.ResourceSelector, apiResources []k8sv1.APIResource) ([]k8sv1.APIResource, error) {
+	var resourceList []k8sv1.APIResource
+	var kindRegexp *regexp.Regexp
+	if filter.KindsRegexp == "" && len(filter.Kinds) == 0 {
+		// if no filters for resource kind are given, return entire resource list
+		return apiResources, nil
+	}
+	if filter.KindsRegexp != "" {
+		var err error
+		kindRegexp, err = regexp.Compile(filter.KindsRegexp)
+		if err != nil {
+			return resourceList, err
+		}
+	}
+	allowedNames := make(map[string]bool)
```

Comment:
nit: might be better to call these `allowedKinds` & `disallowedKinds`.

Reply:
Sure